frb_example/benches/rust/src/api.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(unused_variables)]
+
+use crate::executor::{BenchErrorHandler, BenchExecutor, BenchHandler, Metrics};
+use flutter_rust_bridge::{frb, SyncReturn};
+
+lazy_static::lazy_static! {
+  pub static ref FLUTTER_RUST_BRIDGE_ERROR_HANDLER: BenchErrorHandler = BenchErrorHandler::default();
+  pub static ref FLUTTER_RUST_BRIDGE_HANDLER: BenchHandler = BenchHandler::new(BenchExecutor::new(*FLUTTER_RUST_BRIDGE_ERROR_HANDLER), *FLUTTER_RUST_BRIDGE_ERROR_HANDLER);
+}
+
+/// benchmark time unit
+#[allow(dead_code)] // some variants are used on dart side!
+#[derive(Debug, Clone)]
+pub enum Unit {
+    Milliseconds,
+    Microseconds,
+    Nanoseconds,
+}
+
+/// metric used for [continuous-benchmark](https://github.com/marketplace/actions/continuous-benchmark) worflow
+#[frb]
+#[derive(Debug, Clone)]
+pub struct Metric {
+    pub name: String,
+    pub value: Option<u64>,
+    /// benchmarking time unit is platform-dependent
```

Comment:
about time unit - is it possible we use always `f64` + always a certain unit? For example, suppose the unit is nano, then we seem to be able to represent `(1<<52)/1000000000 = 4503599 seconds` without accuracy loss.

By doing this, we can simplify the code about units.

Reply:
I'm not sure I get what you mean. You want me to refactor time unit to `f64` everywhere ?

Reply:
especially like... what is the benefit ?

Reply:
> I'm not sure I get what you mean. 

`value: u64 + unit` -----> `value: f64`

> what is the benefit

No need to care about time units anymore :) So a lot of code simplification

Reply:
yeah but a lot of time spent for me for a little benefit, don't you think so ?

Reply:
Agree. Would be great if at the beginning we realize this - then can *save* a lot of time; but for now seems ok to leave it like this.

Reply:
I can make the change, but I don't see how `f64` is more appropriate than `u64`. Is it generally faster to compute or something (sorry if it is, I'm not aware of it) ?
I would have thought `usize` instead, which would fit the CPU architecture. Maybe my knowledge is short on this.

Reply:
usize + (for example) always nanosecond also LGTM, since it is big enough to store benchmark values.

f64 comes to my mind just because it accommodates a wide range of values, not necessary the best option, definitely.

Anyway it depends on you whether to change it :)